data-refactoring-advisor/dra-frontend/src/Components/RefineCommunity-Aug14.js

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) 2023, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v1.0 as shown at https://oss.oracle.com/licenses/upl/ 
+
+import React, { useState, useEffect} from "react";
+import { Link } from "react-router-dom";
+import Select from "react-select";
+//import { Link, useNavigate } from "react-router-dom";
+import "../Css/styles.css";
+import { useLocation } from "react-router-dom";
+//import Pagination from "react-responsive-pagination";
+
+function RefineCommunity() {
+
+  const [values, setValues] = useState([]);
+  const [sources, setSources] = useState([]);
+  const [nodes, setNodes] = useState([]);
+
+  const [source, setSource] = useState("");
+  const [destination, setDestination] = useState("");
+  const [destinations, setDestinations] = useState([]);
+  const [destinationArray, setDestinationArray] = useState([]);
+
+  const [affinity, setAffinity] = useState("");
+
+  const [sourceValue, setSourceValue] = useState("");
+  const [destinationValue, setDestinationValue] = useState("");
+
+  const [selectedCard, setSelectedCard] = "";
+  const location = useLocation();
+  //let refineCommunityData =  JSON.parse(location.state.refineCommunityData);
+  let graphData =  JSON.parse(location.state.graphData);
+  let sqlSetName = location.state.stsName;
+
+  useEffect(() => {
+    fetchSources(); 
+  },[]);
+  
+  function fetchSources() {
+
+    const nodes =  JSON.parse(location.state.graphData.nodes);
+    setNodes(nodes);
+
+    /*const sources1 = [...new Set(refineCommunityData.map((item) => item.source))];
+    setSources(sources1)*/
+    const sources1 = [...new Set(graphData.nodes.map((item) => item.name))];
+    
+    /*const newList = values.filter((item) => 
+      sources1.dele
+      item.name !== sourceToRemove
+    );*/
+    
+    //setSources(sources1)
+  }
+
+  function fetchDestinations() {
+    let arrayTemp = [];
+   // GFG["prop_4"] = "val_4";
+    const data = groupBy(graphData.nodes, 'color');
+    const destinations = [...new Set(Object.keys(data).map(objKey => objKey))];
+    destinations.map((item) => {
+      console.log("Dest item == " + item);
+      arrayTemp.push({"color":item})
+    })
```

Comment:
Please check if you can:
Add a "return" statement to this callback.
Callbacks of array methods should have return statements [javascript:S3796](https://sonarcloud.io/organizations/oracle/rules?open=javascript%3AS3796&rule_key=javascript%3AS3796)